test/NuGet.Core.Tests/NuGet.LibraryModel.Tests/CentralPackageVersionNameComparerTests.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Versioning;
+using Xunit;
+
+namespace NuGet.LibraryModel.Tests
+{
+    public class CentralPackageVersionNameComparerTests
+    {
+        [Fact]
+        public void DistinctListOfObjects()
+        {
+            // Arrange
+            var name1 = "name1";
+            var name2 = "name2";
+            var nameU1 = "Name1";
+            var versionRange1 = VersionRange.Parse("1.0.0");
+            var versionRange2 = VersionRange.Parse("2.0.0");
+            var versionRange3 = VersionRange.Parse("3.0.0");
+
+            var cpv11 = new CentralPackageVersion(name1, versionRange1);
+            var cpv12 = new CentralPackageVersion(name1, versionRange2);
+            var cpvU13 = new CentralPackageVersion(nameU1, versionRange3);
```

Comment:
What if you pair nameU1 with versionRange1 ? Would that be distinct?
Your test makes me think that they might only be considered distinct because version 1.0.0 != 3.0.0.

Reply:
cpvU13  is not distinct form the ones above. In the Asserts below is asserted two names "name1" and "name2"`